tsweb/tsweb.go

```diff
@@ -437,12 +439,13 @@ func (h retHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 
 	if bs := h.opts.BucketedStats; bs != nil && bs.Finished != nil {
-		// Only increment metrics for buckets that result in good HTTP statuses.
+		// Only increment metrics for buckets that result in good HTTP statuses
+		// or when we know the start was already counted.
 		// Otherwise they get full of internet scanning noise. Only filtering 404
 		// gets most of the way there but there are also plenty of URLs that are
 		// almost right but result in 400s too. Seem easier to just only ignore
 		// all 4xx and 5xx.
-		if msg.Code < 400 {
+		if startRecorded || msg.Code < 400 {
```

Comment:
do we need the status code check here at all?

Reply:
yes.  This original logic is there to prevent us filling up our stats with bogus requests for URLs like `/wp-login.php` or whatever.  We will want to block creating the bucket on at least one non-error response.

Reply:
so without this status check, and only the `startRecorded` check, we would never actually end up recording anything.  Since the first request to pass this check is the one that actually creates the Started bucket.

Reply:
oh, i see, we don't always increment starts